### ISSUE TITLE
is this keras implementation of coarse temporal attention correct?

### ISSUE THREAD

**Issue by rohitsaluja22** [ArdhenduBehera/cta-net#3]
Hi @ArdhenduBehera, I have implemented coarse temporal attention in keras as follows:
```
 hidden_size = 16
 
def get_attention_focused_hidden_state(self, units, inputs, hidden):
	"""
	    Method implements attention cell.
	"""
	hidden1 = tf.expand_dims(hidden, 1)#?X1X16

	#equation 2 part 3 for psi_t_t'
	confidence = tanh(
	    Dense(units,use_bias=False)(inputs)+Dense(units,use_bias=True)(hidden1)#ψt,t′ = tanh(Wψ ht + ...
	)
	
	#equation2 rhs:-
	beta_t_tdash = sigmoid(Dense(1,use_bias=True)(confidence))#βt,t′ = σ(Wg ψt,t′ + bg ) t′ is 1 to 15 #1X15
	
	#equation2 lhs:-
	at = hidden + tf.reduce_sum(beta_t_tdash * inputs, axis=1) #at = ht +T∑t′ =1βt,t′ ht′  # beta_t_tdash
	return at
	
	
def rnn(self, inputs):
	"""
	    Methods implements gaze lstm.
	"""
	lstm_out_final = TimeDistributed(Masking(mask_value=0), name='dense_final')(inputs)
	lstm_out_final, hidden_state, _ = LSTM(
	    hidden_size, kernel_initializer=self.kernel_initializer, return_sequences=True,
	    recurrent_dropout=self.rec_dropout,
	    bias_initializer=self.bias_initializer,
	    name='lstm_final',
	    return_state=True
	)(lstm_out_final)
	lstm_out_final = Dropout(self.dropout)(lstm_out_final)
	
	ats = []	
	for i in range(TIME_STEPS):
	    at = self.get_attention_focused_hidden_state(3, lstm_out_final, lstm_out_final[:, i])
	    ats.append(at)
	ats = tf.convert_to_tensor(ats)#15XNoneX16 #[15,?,16]
	ats = tf.transpose(ats,[1,0,2]) #NoneX512X15
	
	#equation 3 rhs
	wt = softmax(
	    Dense(1,use_bias=True)(ats),
	    axis=1
	)

	#equation 3 lhs
	op = tf.reduce_sum(wt * ats, axis=1)
	return op

```
I am using this in a model, but the results are not improving. Infact they go much down as compared to Bahdanau's temporal attention. I am confused about equation 3, is W_phi shared for all at's softmax input used to find wt? 

reference: Bahdanau et al. Neural Machine Translation by Jointly Learning to Align and Translate. International Conference on Learning Representations, 2015.
